### PR TITLE
changes made

### PR DESCRIPTION
**Pull request by Anushka-Bhowmick**
## What does this PR do?
Added a new source in competitive programming file | Deleted a resource link which doesn't seem to exist

## For resources
Added Striver's A2Z DSA Course Sheet

## Why is this valuable (or not)?
 It updates the repository with fresh changes

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by Anushka-Bhowmick** [comment]
i have removed the trailing slash yet why is build failing?

File: more/problem-sets-competitive-programming.md [hunk 1]
```diff
@@ -34,6 +33,7 @@
 * [DMOJ](https://dmoj.ca)
 * [E-olymp](https://www.e-olymp.com/en/)
 * [Exercism](https://exercism.org)
+* [GeeksforGeeks](https://practice.geeksforgeeks.org)
```
Comment (eshellman) [inline more/problem-sets-competitive-programming.md:36]: we have 7 geeksforgeeks links already, do we need another?

Comment (Anushka-Bhowmick) [reply more/problem-sets-competitive-programming.md:36]: Sorry! I didn't check if it was available on other files. As it was not mentioned in the practice set problems, so i thought of adding it.

Comment (Anushka-Bhowmick) [reply more/problem-sets-competitive-programming.md:36]: Shall i edit by removing the gfg link as i have also deleted a source (binary search) which doesn't seem to exist?

Comment (eshellman) [reply more/problem-sets-competitive-programming.md:36]: yes, please work on this branch

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**